source/extensions/quic_listeners/quiche/envoy_quic_alarm_factory.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+// QUICHE allows unused parameters.
+#pragma GCC diagnostic ignored "-Wunused-parameter"
```

Comment:
Can you do `GCC diagnostic push` / `pop` respectively before and after quiche imports? So the check still applies to envoy code base.

Reply:
done